### PR TITLE
SEQNG-65: Upgrade to http4s 0.15.0

### PR DESCRIPTION
**Pull request by cquiroz**
This PR started simply as an upgrade to the latest version of http4s but it grew as tests have been added that showed a few bugs

the latest version of http4s changes the way authentication is performed so we should attempt to do it but it showed a harder problem than planned. However the current system still works

This exercise showed some of the limitations of http4s, leading to at least one PR submitted to the project

### CODE REVIEW

**Comment by jdnavarro** [comment]
This comment is not really a review but more just me thinking aloud after checking this PR.

I was just wondering if the JWT code in addition to the associated cookie management code could be encapsulated in a separate module in either `http4s` directly if they are interested in upstreaming it or in a separate package, not related to `ocs3`, if they are not. I'm pretty sure there'll be many people outside of Gemini interested in helping maintain, test and contribute back to a package that facilitates JWT authentication.

**Comment by cquiroz** [comment]
Yes my plan is to divest it into http4s at some moment, I just need to sort out how to make it compatible with the latests http4s Auth Middleware

File: modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.server/src/test/scala/edu/gemini/seqexec/web/server/http4s/SeqexecUIApiRoutesSpec.scala [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      val b = emit(ByteVector.view(Pickle.intoBytes(UserLoginRequest("telops", "pwd"))))
+      for {
+        loginResp    <- OptionT(service.apply(Request(method = Method.POST, uri = uri("/seqexec/login"), body = b)).map(Option.apply))
+        cookieHeader = loginResp.headers.find(_.name === "Set-Cookie".ci)
+        setCookie    <- OptionT(Task.now(cookieHeader.flatMap(u => `Set-Cookie`.parse(u.value).toOption)))
+        authCookie   = new CookieHeader(nels(setCookie.cookie))
+        logoutResp   <- OptionT(service.apply(Request(method = Method.POST, uri = uri("/seqexec/logout")).putHeaders(authCookie)).map(Option.apply))
+      } yield logoutResp
+    }
+}
```
Comment (jdnavarro) [inline modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.server/src/test/scala/edu/gemini/seqexec/web/server/http4s/SeqexecUIApiRoutesSpec.scala]: This must have been a pain to write 👏

File: modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.server/src/main/scala/edu/gemini/seqexec/web/server/security/authentication.scala [hunk 2]
```diff
@@ -62,7 +64,7 @@ case class AuthenticationService(config: AuthenticationConfig) extends AuthServi
     */
   def buildToken(u: UserDetails): String =
     // Given that only this server will need the key we can just use HMAC. 512-bit is the max key size allowed
-    Jwt.encode(JwtClaim(write(u)).issuedNow.expiresIn(3600), config.secretKey, JwtAlgorithm.HmacSHA256)
+    Jwt.encode(JwtClaim(write(u)).issuedNow.expiresIn(sessionTimeout.toSeconds.toLong), config.secretKey, JwtAlgorithm.HmacSHA256)
```
Comment (jdnavarro) [inline modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.server/src/main/scala/edu/gemini/seqexec/web/server/security/authentication.scala]: I'm guessing the default is timeout is still 1 hour, right?

File: modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.server/src/test/scala/edu/gemini/seqexec/web/server/http4s/SeqexecUIApiRoutesSpec.scala (hunk 1 above)
Comment (cquiroz) [inline modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.server/src/test/scala/edu/gemini/seqexec/web/server/http4s/SeqexecUIApiRoutesSpec.scala]: Yes :D but worth the effort. it uncovered a bug

File: modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.server/src/main/scala/edu/gemini/seqexec/web/server/security/authentication.scala (hunk 2 above)
Comment (cquiroz) [inline modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.server/src/main/scala/edu/gemini/seqexec/web/server/security/authentication.scala]: It is configurable with `AuthConfig`

**Pull request merged**